server/neptune/gateway/event/comment_handler.go

```diff
@@ -68,14 +71,29 @@ func (p *CommentEventWorkerProxy) Handle(ctx context.Context, request *http.Buff
 		return p.forwardToSns(ctx, request)
 	}
 
-	if shouldAllocate && command.ForceApply {
-		p.logger.InfoContext(ctx, "running force apply command")
-		if commentErr := p.vcsClient.CreateComment(event.BaseRepo, event.PullNum, warningMessage, ""); commentErr != nil {
-			p.logger.ErrorContext(ctx, commentErr.Error())
+	if shouldAllocate {
+		if cmd.ForceApply {
+			p.logger.InfoContext(ctx, "running force apply command")
+			if commentErr := p.vcsClient.CreateComment(event.BaseRepo, event.PullNum, warningMessage, ""); commentErr != nil {
+				p.logger.ErrorContext(ctx, commentErr.Error())
+			}
+			return p.scheduler.Schedule(ctx, func(ctx context.Context) error {
+				return p.forceApply(ctx, event)
+			})
 		}
-		return p.scheduler.Schedule(ctx, func(ctx context.Context) error {
-			return p.forceApply(ctx, event)
-		})
+
+		// comment legacy apply command message and return
+		if cmd.Name == command.Apply {
+			p.logger.InfoContext(ctx, "invalid legacy apply command on platform mode", map[string]interface{}{
+				"repository": event.BaseRepo.FullName,
+				"pull-num":   event.Pull.Num,
+			})
+			if commentErr := p.vcsClient.CreateComment(event.BaseRepo, event.PullNum, legacyApplyCommandMessage, ""); commentErr != nil {
+				p.logger.ErrorContext(ctx, commentErr.Error())
+			}
+			return nil
```

Comment:
This is incorrect, we shouldn't return because then we're not passing the apply status check, you should let the request get forwarded.